projects/components/src/form/base-form-control.ts

```diff
@@ -102,6 +102,11 @@ export class BaseFormControl<V> implements ControlValueAccessor, CanBeReadOnly {
      */
     @Input() isResponsive = true;
 
+    /**
+     * Aria label to be set when label input is not set
+     */
+    @Input() ariaLabel: string;
```

Comment:
We must provide either `ariaLabel` or `label`. That's a pattern that's good to avoid. Initially I was going to ask that you document the `label` to know that you must provide the `ariaLabel` if you don't provide this. Maybe we should error if we detect this error condition in `ngOnInit` . Then I started trying to think of a way to avoid the problem altogether.

if instead, we added a new `@Input() hideLabel=false`? Then it just uses the label input as the `aria-label`? It makes sure someone doesn't forget to set either and shields the user from needing to understand about aria.

Reply:
Angular 16 is going to allow for required inputs, and label should be required. It can currently be hacked to be required. See https://indepth.dev/posts/1517/required-inputs-in-angular-v16

Reply:
I also had the same thoughts considering something like `hideLabel`, but I did not like that my idea because I have a `label` input and I would expect it won't be displayed if I pass empty string or just do not pass it. And that `hideLabel` sounds strange when users can just not pass the label. That's why I ended up with that aria-label. I also thought that we may have some eslint rule that will check that ariaLabel is set when no label is passed - this will ensure all form controls will meet the a11y requirement. 

Now that you are thinking also in that direction makes me think that this could be the better approach. In addition to that we may make the label as required param when we upgrade to angular 16. But it still looks odd to me that we are asking for a label and having hideLabel=true.

Let me know if you insist in the `hideLabel` input and I will update the code.

Reply:
I like it because:
*  We can enforce that a label must always provided. We can enforce it even before v16 as https://indepth.dev/posts/1517/required-inputs-in-angular-v16  shows.
* The user doesn't need to understand about aria to get a11y goodness. The code will guide them that they must pass in a label. The documentation can easily state that if you don't want it to be displayed, you can set it to hidden using `hideLabel`. Maybe we don't even need a new `hideLabel` we can just make label be of type `string |  HiddenLabel`.


Feel free to ask for feedback from others.

Reply:
As we agreed in our internal Slack discussion and poll - we will have a new `@Input hideLabel` that will be used for hiding the label. We will assume the `label` value is always set until we migrate to Angular 16 and be able to make it required.
Changes have been applied and this merge request could be reviewed again.